finality-aleph/src/metrics.rs

```diff
@@ -71,11 +71,25 @@ pub(crate) enum Checkpoint {
 
 #[derive(Clone)]
 pub struct Metrics<H: Key> {
-    inner: Arc<Mutex<Inner<H>>>,
+    inner: Option<Arc<Mutex<Inner<H>>>>,
 }
 
 impl<H: Key> Metrics<H> {
-    pub fn register(registry: &Registry) -> Result<Self, PrometheusError> {
+    pub fn new(registry: Option<&Registry>) -> Metrics<H> {
```

Comment:
I don't think it's particularly great to have logging happening here. I saw the discussion about `Result`, but I still think returning a result here is better. To keep the log about running without metrics I would add a `is_noop` method to `Metrics` and invoke it in the one place where we care. At the same time the error could be logged and the result produced using the `noop` constructor I suggested above.